XLS-47d-PriceOracles/README.md

```diff
@@ -0,0 +1,453 @@
+
+### Terminology
+
+- Oracle Provider is a service or technology that enables the integration of external data and real-world events into a blockchain network.
+- dApp, short for decentralized application, refers to an application that is built on a blockchain network and operates using smart contracts or rely on other mechanisms or protocols for their functionality.
+
+## Creating `PriceOracle` instance on XRPL
+
+### On-Ledger Data Structure
+
+#### The `PriceOracle` Object
+
+The `PriceOracle` ledger entry represents the `PriceOracle` object on XRPL ledger and contains the following fields:
+
+|FieldName | Required? | JSON Type | Internal Type|
+|:---------|:-----------|:---------------|:---------------|
+| `LedgerEntryType`| :heavy_check_mark: | `string` | `UINT16` |
+| `Owner` | :heavy_check_mark: | `string` | `ACCOUNTID` |
+| `Provider` | :heavy_check_mark: | `string` | `BLOB` |
+| `PriceDataSeries` | :heavy_check_mark: | `array` | `ARRAY` |
+| `LastUpdateTime` | :heavy_check_mark: | `number` | `UINT32` |
+| `URI` | | `string` | `BLOB` |
+| `SymbolClass` | :heavy_check_mark: | `string` | `BLOB` |
+| `PreviousTxnID` | :heavy_check_mark: | `string` | `UINT256` |
+| `PreviousTxnLgrSeq`| :heavy_check_mark: | `string` | `UINT32` |
+
+- `LedgerEntryType` identifies the type of ledger object. The proposal recommends the value 0x0080 as the reserved entry type.
+- `Owner` is the account that has the update and delete privileges. It is recommended that this account has an associated [`signer list`](https://xrpl.org/set-up-multi-signing.html).
+- `Provider` identifies an Oracle Provider. It can be URI or any data, for instance `chainlink`. It is a string of up to 256 ASCII hex encoded characters (0x20-0x7E).
+- `PriceDataSeries` is an array of up to ten `PriceData` objects, where `PriceData` represents the price information for a token pair. `PriceOracle` with more than five `PriceData` objects requires two owner reserves. `PriceData` includes the following fields:
+
+  |FieldName | Required? | JSON Type | Internal Type|
+  |:---------|:-----------|:---------------|:---------------|
+  | `Symbol` | :heavy_check_mark: | `string` | `CURRENCY` |
+  | `PriceUnit` | :heavy_check_mark: | `string` | `CURRENCY` |
```

Comment:
Say in a `BTC/USD` trading pair, `BTC` is the `Symbol` and `USD` is the `PriceUnit`. For instance a price of 36,180 for a `BTC/USD` means that one unit of `BTC` valued at 36,180 units of `USD`. How about changing these two fields to the following to provide more clarity:

- `BaseAsset`: This refers to the primary asset within a trading pair. It is the asset against which the price of the quote asset is quoted. The base asset is usually considered the 'primary' asset and forms the basis for trading. For example:
In the pair `BTC/USD`, `BTC` is the base asset.
In `ETH/BTC`, `ETH` is the base asset.

- `QuoteAsset`: This represents the secondary or quote asset in a trading pair. It denotes the price of one unit of the base asset. The quote asset's value is expressed in terms of the base asset. For example:
In the pair `BTC/USD`, `USD` is the quote asset.
In `ETH/BTC`, `BTC` is the quote asset.
The base asset and quote asset together form a trading pair, and their relationship determines the price at which one asset can be exchanged for another.

The `CURRENCY` can be changed to `ASSET_TYPE` to give a more generalized meaning. The `ASSET_TYPE` is either the three-letter currency or some other asset represented as a hexadecimal string. Either way it's a 160-bit field, which is conformant to the XRPL currency description: https://xrpl.org/currency-formats.html#currency-codes.

There is also a `SymbolClass` field, which provides additional information for the `BaseAsset`. I'm proposing to change it to `AssetClass` to reflect the above updates.

Reply:
Yes and yes. Agree with everything - what you wrote above is much clearer!

Reply:
@sappenin I think we should keep the `CURRENCY` because the internal type is `Currency` and it can't be changed unless a duplicate type `AssetType` is introduced behaving exactly like the `Currency`. This doesn't make much sense. Another option is use the `VL` (blob) type of say up to 20 bytes. The goodness of the `Currency` is the standard currency (3-letters) can be passed as a regular string but the drawback is that anything else has to be a 160-bit hexadecimal, so we are going to waste some space. The advantage of the `VL` is that it's a variable length so no space is wasted but the drawback is that all input has to be passed as a hexadecimal.

Reply:
+1 on keeping `CURRENCY`. 

Unless I missed anything, I think this comment can be resolved @gregtatcam once you've captured everything useful from it.